cmd/agent/common/autoconfig_nodocker.go

```diff
@@ -8,14 +8,40 @@
 package common
 
 import (
+	"path/filepath"
+
+	"github.com/DataDog/datadog-agent/pkg/collector"
+	"github.com/DataDog/datadog-agent/pkg/collector/autodiscovery"
+	"github.com/DataDog/datadog-agent/pkg/collector/loaders"
+	"github.com/DataDog/datadog-agent/pkg/collector/providers"
 	log "github.com/cihub/seelog"
 )
 
-// SetupAutoConfig placeholder if docker is disabled
+// SetupAutoConfig only starts the Autoconfig subsystem if Docker is not available
 func SetupAutoConfig(confdPath string) {
-	log.Debugf("AutoDiscovery is only supported with docker, disabling")
+	// create the Collector instance and start all the components
+	// NOTICE: this will also setup the Python environment
+	Coll = collector.NewCollector("")
+
+	// create the Autoconfig instance
+	AC = autodiscovery.NewAutoConfig(Coll)
+
+	// add the check loaders
+	for _, loader := range loaders.LoaderCatalog() {
+		AC.AddLoader(loader)
+		log.Debugf("Added %s to AutoConfig", loader)
+	}
+
+	// Add the configuration providers
+	// File Provider is hardocded and always enabled
+	confSearchPaths := []string{
+		confdPath,
+		filepath.Join(GetDistPath(), "conf.d"),
+	}
+	AC.AddProvider(providers.NewFileConfigProvider(confSearchPaths), false)
```

Comment:
why not still use the `ProviderCatalog` here? (unless I'm missing something, if the build tags are defined correctly everywhere the catalog should reflect what the agent was built with)

I completely understand if you don't want to spend time to improve this now, let's just add a `FIXME`/`TODO` in that case

Reply:
This is even bigger than that, configuration providers should not be tied to `docker`. A legit use case would be using `etcd` to store configurations without polling it, because you don't use Autodiscovery, or we might add new providers in the future that work with or without Autodiscovery. For the time being I thought it was more straightforward to just hardcode the file configurations provider, will add a code comment to recap all of this.